timelock_auction/vickrey_auction/lib.rs

```diff
@@ -194,66 +138,59 @@ mod vickrey_auction {
                 self.participants.push(bidder);
             }
 
-            self.proposals.insert(bidder, 
-                &Proposal {
-                    ciphertext, 
-                    nonce, 
-                    capsule,
-                    commitment,
-                });
-            Self::env().emit_event(BidSuccess{});
+            Self::env().emit_event(BidSuccess {});
             Ok(())
         }
 
-          /// complete the auction
-          /// 
-          /// * `revealed_bids`: A collection of (participant, revealed_bid_amount)
-          ///
-          #[ink(message)]
-          pub fn complete(
-              &mut self, 
-              revealed_bids: Vec<RevealedBid<AccountId>>,
-          ) -> Result<(), Error> {
+        /// Takes de incoming reveled bid and saves it in the revealed_bids array
+        ///
+        /// * `revealed_bid`: the revealed bid
+        ///
+        #[ink(message)]
+        pub fn save_revealed_bid(
+            &mut self,
+            revealed_bid: RevealedBid<AccountId>,
+        ) -> Result<(), Error> {
+            let who = self.env().caller();
+            if who != self.proxy {
+                return Err(Error::NotProxy);
+            }
+            if !self.participants.contains(&revealed_bid.bidder.clone()) {
+                return Err(Error::NotParticipant);
+            }
+            self.revealed_bids.push(RevealedBid {
+                bidder: revealed_bid.bidder,
+                bid: revealed_bid.bid,
+            });
+            Ok(())
+        }
+
+        /// Complete the auction
+        /// Checks the revealed bids and determines the winner
+        ///
+        #[ink(message)]
+        pub fn complete(&mut self) -> Result<(), Error> {
+            if self.revealed_bids.len() != self.participants.len() {
```

Comment:
We can get rid of this check. If anybody doesn't participate we should just ignore them and continue, so if there are less bids than committed players it should be fine, the contract just gets some free tokens if a player doesn't  bid